### ISSUE TITLE
Jaeger - namespace_selector not working for services in istio-system

### ISSUE THREAD

**Issue by davtex** [kiali/kiali#4935]
Internal query to Jaeger and sometimes external query to Jaeger is missing namespace if the selected service/application/workload is in istio-system namespace despite "namespace_selector" being set to "true".

I do not have issues with any other namespaces, e.g. developer-poratl-uat -> I always have "service=service.namespace" in query:
```
Application:
https://jaeger.infra.dc.uat.k8s.XYZ/search?limit=100&service=devportaluseraccountmfe.developer-portal-uat&start=1649304127750000
http://jaeger-query.jaeger:16686/api/traces?end=1649325727000000&limit=100&service=devportaluseraccountmfe.developer-portal-uat&start=1649304127000000

Workload:
https://jaeger.infra.dc.uat.k8s.XYZ/search?limit=100&service=devportaluseraccountmfe.developer-portal-uat&start=1649304174563000
http://jaeger-query.jaeger:16686/api/traces?end=1649325910000000&limit=500&service=devportaluseraccountmfe.developer-portal-uat&start=1649325310000000

Services:
https://jaeger.infra.dc.uat.k8s.XYZ/search?service=devportaluseraccountmfe.developer-portal-uat&start=1649304400096000&limit=100
http://jaeger-query.jaeger:16686/api/traces?end=1649326011000000&limit=100&service=devportaluseraccountmfe.developer-portal-uat&start=1649325411000000
```

Problem is only with istio-system, where namespace is always missing with exception of Application's external query:
```
Application:
https://jaeger.infra.dc.uat.k8s.XYZ/search?limit=100&service=istio-ingressgateway.istio-system&start=1649304450753000
http://jaeger-query.jaeger:16686/api/traces?end=1649326141000000&limit=100&service=istio-ingressgateway&start=1649304541000000

Workload:
https://jaeger.infra.dc.uat.k8s.XYZ/search?service=istio-ingressgateway&start=1649304627157000&limit=100
http://jaeger-query.jaeger:16686/api/traces?end=1649326227000000&limit=500&service=istio-ingressgateway&start=1649304627000000

Services:
https://jaeger.infra.dc.uat.k8s.XYZ/search?limit=100&service=istio-ingressgateway&start=1649304671673000
http://jaeger-query.jaeger:16686/api/traces?end=1649326331000000&limit=100&service=istio-ingressgateway&start=1649304731000000
```

I've tried moving istio-ingressgateway to different namespace to eliminate possibility that it's some labeling issue or  something and it worked -> as soon as I moved it to "kiali" namespace, query had "service=istio-ingressroute.kiali" in it. So as far as I can say, "namespace_selector" functionality is somehow broken for services inside namespace defined in "istio_namespace" Kiali config.

I have also tried removing my label selector, "whitelist_istio_system" parameter, component_status etc. -> it did not have any impact. Only solution was to move istio-ingressgateway to different NS.

I am using v1.48 Kiali Operator with following Kiali CR:
```
apiVersion: kiali.io/v1alpha1
kind: Kiali
metadata:
  annotations:
  generation: 1
  labels:
    argocd.argoproj.io/instance: kiali
  name: kiali
  namespace: istio-system
spec:
  api:
    namespaces:
      label_selector: istio-injection=enabled
  auth:
    strategy: anonymous
  deployment:
    accessible_namespaces: '**'
    image_name: XYZ/kiali/kiali
    image_pull_secrets:
    - harbor-secret
    image_version: v1.48.0
    ingress:
      class_name: infra-traefik
      enabled: true
    namespace: istio-system
    pod_labels:
      sidecar.istio.io/inject: "false"
    resources:
      requests:
        cpu: 50m
        memory: 64Mi
  external_services:
    grafana:
      enabled: false
    istio:
      component_status:
        components:
        - app_label: istiod
          is_core: true
          is_proxy: false
        - app_label: istio-ingressgateway
          is_core: true
          is_proxy: true
        - app_label: istio-eastwestgateway
          is_core: true
          is_proxy: true
        enabled: true
    prometheus:
      auth:
        insecure_skip_verify: true
      url: https://prometheus.infra.dc.uat.k8s.XYZ
    tracing:
      enabled: true
      in_cluster_url: http://jaeger-query.jaeger:16686
      namespace_selector: true
      url: https://jaeger.infra.dc.uat.k8s.XYZ
      use_grpc: false
      whitelist_istio_system:
      - jaeger-query
      - istio-ingressgateway
      - istio-eastwestgateway
  istio_labels:
    app_label_name: app
    version_label_name: version
  istio_namespace: istio-system
  server:
    web_fqdn: kiali.infra.dc.uat.k8s.XYZ
    web_root: /
    web_schema: https
```

istioctl:
client version: 1.12.2
control plane version: 1.13.2
data plane version: 1.13.2 (90 proxies)
kubectl:
Client Version: v1.22.5
Server Version: v1.20.4

**Comment by lucasponce** [comment]
Thanks, @davtex for the issue.
Yes the "istio-system" (control plane) has special handling, which services/workloads are being deployed therefrom your example? More Gateways? Or do you have any other workload that should be part of the trace and it's failing for some reason?

Well, I think that it shouldn't fail, at least if no data, it should return some empty result unless there is something wrong.

Could you confirm if you are using the jaeger version shipped with the Istio addons?

**Comment by davtex** [comment]
@lucasponce I have istiod, istio-ingressgateway, istio-egressgateway and Kiali deployed in "istio-system" namespace. I am mainly interested in seeing istio-ingressgateway Traces using Kiali, so a simple use-case.

Yes the query itself is valid and not failing, but since the "service" is incomplete (should be istio-ingressgateway.istio-system), I get 0 results. It's not anything major but I like Kiali's internal Trace viewer and it's a shame it's not working in this case.

I am using Jaeger Operator to deploy simple in-memory Jaeger -> I am NOT using one provided by Istio.

**Comment by lucasponce** [comment]
> It's not anything major but I like Kiali's internal Trace viewer and it's a shame it's not working in this case.

Thanks for the clarification, probably how Kiali builds the "jaeger" service name can be reviewed to see what's missing for this scenario.

**Comment by davtex** [comment]
@lucasponce 

Found it, just do not understand why it is there:
`func buildJaegerServiceName(namespace, app string) string {
	conf := config.Get()
	if conf.ExternalServices.Tracing.NamespaceSelector && namespace != conf.IstioNamespace {
		return app + "." + namespace
	}
	return app`

**Comment by jshaughn** [comment]
@lucasponce This has gone stale, did you have any feedback for the latest reply?

**Comment by israel-hdez** [comment]
I traced back when that conditional was introduced. It was here, 2.5 years ago: #2017.

Also for me it is not clear why the `&& namespace != conf.IstioNamespace` part is there, but that is because I not familiar on how Kiali's tracing integration works.

Fortunately, @aljesusg is still here and may comment. Perhaps, after 2.5 years is hard to remember why that conditional was introduced, but @aljesusg has much more familiar with the Jaeger integration and may tell if it is safe to remove that conditional or not.

**Comment by jshaughn** [comment]
Thanks @israel-hdez for commenting, this one had somehow fallen off the radar.  Is this possibly related to the problem being discussed in https://github.com/kiali/kiali/discussions/5513?  @aljesusg?

**Comment by aljesusg** [comment]
Well let me explain this.

Some users use the terminology **service_name.namespace** for tracing, the point is that when we developed this **Istio installation** did not add the namespace to the services under istio-system. This is why we added this condition. i don't know the picture now. But it seems after seeing @davtex comments this is now working like before, right?

**Comment by israel-hdez** [comment]
@jshaughn I just dropped a comment on #5513. IMO, the issues are unrelated.

> @aljesusg But it seems after seeing @davtex comments this is now working like before, right?

I'm not sure to understand this sentence :-)
What I understand is that @davtex had to do a workaround to make Kiali work. But things does not work in his desired setup.

**Comment by aljesusg** [comment]
Wow it's wrong 😅 I mean that after read all the comments it seems that this is NOT working like before we should investigate it

**Comment by davtex** [comment]
> Well let me explain this.
> 
> Some users use the terminology **service_name.namespace** for tracing, the point is that when we developed this **Istio installation** did not add the namespace to the services under istio-system. This is why we added this condition. i don't know the picture now. But it seems after seeing @davtex comments this is now working like before, right?

This is no longer true as far as I can say. Istio is submitting traces from kube-system with namespace included, same as with any other namespace. Kiali is omitting the namespace in query, resulting in no results.

**Comment by israel-hdez** [comment]
OK, I'll do a few tests.
If my testing show the same issue as yours, most likely it is safe to remove that condition.

@davtex In the meantime, would you mind submitting a PR with the needed change to fix it? Looks like a simple change and you get your name in the [contributors list](https://github.com/kiali/kiali/graphs/contributors).

That said, I'm not sure the fix will be back-ported to the version of Kiali that you are using :-/

**Comment by israel-hdez** [comment]
I can confirm that Traces for the `istio-ingressgateway` are not visible in Kiali.

Weirdly, the front-end generates the right link. So, if you click the "View in Tracing" link, it will open the right view:

![TracesBug](https://user-images.githubusercontent.com/23639005/193874897-1951731d-e289-4c4b-a2f6-514a790a0b39.gif)

So, only the query that Kiali is doing is problematic.
I'll wait a PR from @davtex. If it is not there by mid next week, I'll fix to have this working in the next release.

@jshaughn Probably, the fix will need to be back-ported to v1.57, right?

**Comment by israel-hdez** [comment]
BTW, also the graph's sidepanel is affected. It does not show the traces list when selecting a workload of the `istio-system` namespace.
